src/tools/suggest-tests/src/tests.rs

```diff
@@ -0,0 +1,21 @@
+macro_rules! sugg_test {
+    ( $( $name:ident: $paths:expr => $suggestions:expr ),* ) => {
+        $(
+            #[test]
+            fn $name() {
+                let suggestions = crate::get_suggestions(&$paths).into_iter().map(|s| s.to_string()).collect::<Vec<_>>();
+                assert_eq!(suggestions, $suggestions);
+            }
+        )*
+    };
+}
+
+sugg_test! {
+    test_error_code_docs: ["compiler/rustc_error_codes/src/error_codes/E0000.md"] =>
+        ["check N/A", "test compiler/rustc_error_codes N/A", "test linkchecker 0", "test src/test/ui src/test/run-make 1"],
+
+    test_rustdoc: ["src/librustdoc/src/lib.rs"] => ["test rustdoc 1"],
+
+    test_rustdoc_and_libstd: ["src/librustdoc/src/lib.rs", "library/std/src/lib.rs"] =>
+        ["test library/std 0", "test rustdoc 1"]
+}
```

Comment:
nit: this would be easier to read if you wrote it as a for loop inside a single `#[test]` function instead of a macro.

Reply:
Personally I like the idea of having one test per suggestion, it's clearly structured and easier to tell what test failed. Also I think it's quite clear to read, it's just a function with placeholders effectively.

Reply:
The macro could generate a single function, with a for loop.

Reply:
But then you don't get the nice "function `test_suggestion_xyz` failed"?